Python/specialize.c

```diff
@@ -1794,6 +1812,32 @@ _Py_Specialize_CompareOp(PyObject *lhs, PyObject *rhs,
             goto success;
         }
     }
+#ifdef Py_STATS
+    if (PyBytes_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_BYTES);
+        goto failure;
+    }
+    if (PyTuple_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_TUPLE);
+        goto failure;
+    }
+    if (PyList_CheckExact(lhs)) {
+        SPECIALIZATION_FAIL(COMPARE_OP, SPEC_FAIL_COMPARE_LIST);
+        goto failure;
+    }
+    if (PySet_CheckExact(lhs) || PyFrozenSet_CheckExact(lhs)) {
```

Comment:
```suggestion
    if (PyAnySet_CheckExact(lhs)) {
```

Reply:
I'd personally rather not -- IMO the `||` is more clear and self-explanatory.